packages/common-ui-web/src/components/VehiclePartSelection.tsx

```diff
@@ -0,0 +1,93 @@
+import { PartSelectionOrientation, VehiclePart, VehicleType } from '@monkvision/types';
+import { useEffect, useState } from 'react';
+import { useMonkTheme } from '@monkvision/common';
+import { Icon } from '../icons';
+import { VehicleDynamicWireframe, VehicleDynamicWireframeProps } from './VehicleDynamicWireframe';
+
+export interface VehiclePartSelectionProps {
+  vehicleType: VehicleType;
+  orientation?: PartSelectionOrientation;
+  /**
+   * emit when change in the selected parts.
+   */
+  onPartsSelected?: (parts: VehiclePart[]) => void;
+}
+
+/**
+ * Move through different orientations of the vehicle wireframe.
+ * And choose the parts
+ */
+export function VehiclePartSelection({
+  vehicleType,
+  orientation: initialOrientation,
+  onPartsSelected = () => {},
+}: VehiclePartSelectionProps) {
+  const partSelectionOrientations = [
+    PartSelectionOrientation.FRONT_LEFT,
+    PartSelectionOrientation.REAR_LEFT,
+    PartSelectionOrientation.REAR_RIGHT,
+    PartSelectionOrientation.FRONT_RIGHT,
+  ];
+  const [orientation, setOrientation] = useState(
+    initialOrientation ?? partSelectionOrientations[0],
+  );
+  const moveOrientation = (direction: 'next' | 'previous') => {
+    const currentIndex = partSelectionOrientations.indexOf(orientation);
+    const nextIndex =
+      direction === 'next'
+        ? (currentIndex + 1) % partSelectionOrientations.length
+        : (currentIndex - 1 + partSelectionOrientations.length) % partSelectionOrientations.length;
+    setOrientation(partSelectionOrientations[nextIndex]);
+  };
+  const [selectedParts, setSelectedParts] = useState<Array<VehiclePart>>([]);
+  useEffect(() => {
+    onPartsSelected(selectedParts);
+  }, [selectedParts]);
+  const togglePart = (part: VehiclePart) => {
+    if (selectedParts.includes(part)) {
+      setSelectedParts(selectedParts.filter((p) => p !== part));
+    } else {
+      setSelectedParts([...selectedParts, part]);
+    }
+  };
```

Comment:
Instead of a useEffect, directly call the side effect in the event handler :

```typescript
const handleClickPart = (part: VehiclePart) => {
  const newSelectedParts = selectedParts.includes(part)
    ? selectedParts.filter((p) => p !== part)
    : [...selectedParts, part];
  setSelectedParts(newSelectedParts);
  onPartsSelected?.(newSelectedParts);
};
```

Reply:
i still wound recomment to keep onPartsSelected under useEffect because it will give more reactive approach than declerative approach.

Reply:
I understand your point and it is a completely valid point. The issue with having reactive effects all around the components is that in order to understand what is going on, a new-coming developer has to scroll through all of the component's code and analyze every effect in existence to check what is happening. In the previous project, we had huge components that contained a lot of useEffects, and it became hard to understand what is going on by simply looking at the code. But for smaller components your point stands valid, I think it is just a matter of preference.

I'm not resolving the conversation now so that you can read this comment. Once you have read this comment, you can resolve the conversation on your own.